arch/arm/src/common/Toolchain.defs

```diff
@@ -292,8 +320,13 @@ endif
 
 # Architecture flags
 
-ARCHCFLAGS += -Wstrict-prototypes -Wno-attributes -Wno-unknown-pragmas
-ARCHCXXFLAGS += -Wno-attributes -Wno-unknown-pragmas
+ifeq ($(CONFIG_ARM_TOOLCHAIN_GHS),y)
+  ARCHCFLAGS += -gcc -gnu99 -preprocess_assembly_files --diag_suppress=68,111,174,222,236,257,826,1143,1721
+  ARCHCXXFLAGS += --gnu_asm --diag_suppress=540,826
```

Comment:
Add some comments to describe the meaning of diag_suppress numbers, you could refer:
https://github.com/apache/nuttx/blob/master/arch/tricore/src/common/ToolchainTasking.defs#L107-L125